ledger/acctupdates.go

```diff
@@ -1352,17 +1358,27 @@ func (au *accountUpdates) lookupWithoutRewards(rnd basics.Round, addr basics.Add
 			rnd = currentDbRound + basics.Round(currentDeltaLen)
 		}
 
+		if synchronized {
+			au.accountsMu.RLock()
+			needUnlock = true
+		}
 		// check the baseAccounts -
 		if macct, has := au.baseAccounts.read(addr); has {
+			// there is a chance the db advanced between deltas check and the baseAccounts check
+			// ensure the round requested is still in the db
+			if macct.Round > rnd {
+				return ledgercore.AccountData{}, 0, "", 0, &RoundOffsetError{rnd, macct.Round}
```

Comment:
Why do you think it is worth taking an optimistic (instead of preventing an error here with locks), and returning an error when the database advances?
There are operation that need to get canceled because an error is returned from here.

Reply:
This is only actual for rest api since txpool and block validator need the latest state of account to validate.